File: frontend/src/main/java/org/sc/frontend/configuration/AppProperties.java
package org.sc.frontend.configuration;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class AppProperties {

    private final String port;
    private final String address;

    @Autowired
    public AppProperties(final @Value("${server.port}") String port,
                         final @Value("${service.backend.address}") String address) {
        this.port = port;
        this.address = address;
    }

    public String getPort() {
        return port;
    }

    public String getBackendAddress() {
        return address;
    }
}

File: frontend/src/main/java/org/sc/frontend/configuration/AppProperties.kt
package org.sc.frontend.configuration

import javax.inject.Inject
import javax.inject.Named

class AppProperties @Inject constructor(@Named(PORT_PROPERTY) val webPort: Int,
                                        @Named(BACKEND_ADDRESS) val backendAddress: String) {

    companion object {
        private const val PORT_PROPERTY = "web-port"
        private const val BACKEND_ADDRESS = "backend-address"
    }

}